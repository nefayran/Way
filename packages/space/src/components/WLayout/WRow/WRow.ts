import { computed, h, defineComponent } from "vue";

export default defineComponent({
  name: "WRow",

  props: {
    row: {
      type: [Boolean, String, Number],
      default: 1,
    },
    tag: {
      type: String,
      default: "div",
    },
  },

  setup(props, { slots }) {
    const classes = computed(() => {
      const classList: any[] = [];

      classList.push({
        [`row-${props.row}`]: props.row,
      });

      return classList;
    });

    return () =>
      h(
        props.tag,
        {
          class: ["w-row row", classes.value],
        },
        slots.default?.()
      );
  },
});
