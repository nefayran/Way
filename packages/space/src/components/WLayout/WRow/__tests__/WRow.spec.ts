import { mount } from "@vue/test-utils";
import WRow from "../WRow";

describe("WRow.ts", () => {
  test("should have an w-row class", () => {
    const wrapper = mount(WRow, {
      propsData: {
        row: "2",
      },
    });
    expect(wrapper.classes()).toContain("w-row");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
