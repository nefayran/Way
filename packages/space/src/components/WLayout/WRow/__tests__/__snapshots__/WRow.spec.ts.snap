// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WRow.ts should have an w-row class 1`] = `<div class="w-row row row-2"></div>`;
